File: solutions/125_Valid_Palindrome.js
// 125. Valid Palindrome Solution

/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {

    if (!s){
      return true
    }

    let stringStripped = s.toLowerCase().replace(/[^a-z0-9]/g,"").split("").join("")

    let stringReversed = stringStripped.split("").reverse().join("")

    if (stringStripped == stringReversed){
      return true
    }

    return false
    
};

File: solutions/125_Valid_Palindrome.java
// 125. Valid Palindrome JAVA solution
// https://leetcode.com/problems/valid-palindrome/

class Solution {
    public boolean isPalindrome(String s) {
        
        if (s == null){
            return true;
        }

        String cleanedString = s.toLowerCase().replaceAll("[^a-z0-9]", "");

        String reversedString = new StringBuilder(cleanedString).reverse().toString();

        if (cleanedString.equals(reversedString)){
            return true;
        }

        return false;
        
    }
}